easybuild/easyconfigs/m/MOOSE/MOOSE-2021-05-18-foss-2019b-Python-3.7.4.eb

```diff
@@ -0,0 +1,66 @@
+easyblock = 'Binary'
+
+name = 'MOOSE'
+# corresponds to commit f5fa9f0
+version = '2021-05-18'
+versionsuffix = '-Python-%(pyver)s'
+
+homepage = 'https://mooseframework.inl.gov'
+description = """The Multiphysics Object-Oriented Simulation Environment (MOOSE) is a finite-element, multiphysics
+framework primarily developed by Idaho National Laboratory"""
+
+toolchain = {'name': 'foss', 'version': '2019b'}
+toolchainopts = {'usempi': True}
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('libpng', '1.6.37'),
+    ('PETSc', '3.12.4', versionsuffix),
+    ('SLEPc', '3.12.2', versionsuffix),
+    ('VTK', '8.2.0', versionsuffix),
+    ('sympy', '1.5.1', versionsuffix),
+    ('ParMETIS', '4.0.3'),
+]
+
+sources = [{
+    'filename': SOURCE_TAR_GZ,
+    'git_config': {
+        'url': 'https://github.com/idaholab',
+        'repo_name': 'moose',
+        'tag': version,
+        'recursive': True,
+        'keep_git_dir': True,
+    },
+}]
+checksums = [None]
+
+extract_sources = True
+
+buildininstalldir = True
+
+# see https://mooseframework.inl.gov/getting_started/installation/hpc_install_moose.html
+# enable building libmesh with -march=native
+local_libmesg_configopts = "--disable-warnings --enable-march --with-mpi=$EBROOTMPI"
+install_cmd = "sed 's/--disable-warnings/%s/g' scripts/update_and_rebuild_libmesh.sh && " % local_libmesg_configopts
+# build vendored copy of libmesh
+install_cmd += 'export CXXFLAGS="$CXXFLAGS -fpermissive" && export MOOSE_JOBS="%(parallel)s V=1" && '
+install_cmd += "./scripts/update_and_rebuild_libmesh.sh --skip-submodule-update && "
+# build MOOSE itself
+install_cmd += "cd test && pwd && make -j %(parallel)s && make -j %(parallel)s hit && "
+# run tests
+install_cmd += "echo 'running tests' && export PYTHONPATH=%(installdir)s/moose/moosetools/contrib/hit:$PYTHONPATH && "
+# run tests, but tolerate failures
+install_cmd += "(python run_tests -j %(parallel)s || echo 'Some tests are failing!' >&2 ) "
```

Comment:
How many are we tolerating?
```
Ran 1959 tests in 133.1 seconds. Average test time 2.5 seconds, maximum test time 32.2 seconds.
^[[1m^[[32m1909 passed^[[0m, ^[[1m147 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m^[[31m50 FAILED^[[0m
^[[1m^[[31mMAX FAILURES REACHED^[[0m
```

Reply:
I made some effort to look into failing tests, but that seemed to require domain knowledge.

Maybe we should reach out to the MOOSE developers with this?

Reply:
I've looked some more and run the tests without the `-j %(parallel)s` and with `--max-fails 1000000`:
```
Ran 2630 tests in 1621.0 seconds. Average test time 0.5 seconds, maximum test time 25.5 seconds.
^[[1m^[[32m2543 passed^[[0m, ^[[1m156 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m^[[31m87 FAILED^[[0m
Some tests are failing!
```

So 87 of 2630 failing. However, some of the failures are:
```
ERROR: Functionality is not available.
Make sure LIBMESH_HAVE_XDR is defined at build time
The XDR interface is not available in this installation
```

Enabling that (and adding a `libtirpc` dependency) gets me to 42 of 2682 failing. I've no idea why I now have more tests running...

Next I see two types of PETSc errors:

```
ERROR: SCOTCH_dgraphInit: Scotch compiled with SCOTCH_PTHREAD and program not launched with MPI_THREAD_MULTIPLE
PETSC ERROR: #1 MatPartitioningApply_PTScotch_Private() line 337 in /dev/shm/build-branfosj-admin/branfosj-admin-up/PETSc/3.12.4/foss-2019b-Python-3.7.4/petsc-3.12.4/src/mat/partition/impls/scotch/scotch.c
PETSC ERROR: #2 MatPartitioningApply_PTScotch() line 423 in /dev/shm/build-branfosj-admin/branfosj-admin-up/PETSc/3.12.4/foss-2019b-Python-3.7.4/petsc-3.12.4/src/mat/partition/impls/scotch/scotch.c
PETSC ERROR: #3 MatPartitioningApply() line 332 in /dev/shm/build-branfosj-admin/branfosj-admin-up/PETSc/3.12.4/foss-2019b-Python-3.7.4/petsc-3.12.4/src/mat/partition/partition.c
PETSC ERROR: #4 partitionGraph() line 303 in /rds/bear-apps/devel/eb-sjb-up/EL8/EL8-cas/software/MOOSE/2021-05-18-foss-2019b-Python-3.7.4/moose/framework/src/partitioner/PetscExternalPartitioner.C

```

and

```
PETSC ERROR: Unknown type. Check for miss-spelling or missing package: https://www.mcs.anl.gov/petsc/documentation/installation.html#external
PETSC ERROR: Unknown partitioning type parmetis
```

Beyond those I am seeing some numerical errors. It'd be good to know from the MOOSE developers if MOOSE is similar to CP2K ('there will always be failures') or if we should have all the tests pass.